indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
                 if self._useColors():
-                    l = [["",p1]]
-                    widths=[0.2*cm,None]
                     ts = deepcopy(colorts)
-                    ts.add('BACKGROUND', (0, 0), (0,-1), self._getSessionColor(sessionSlot))
-                    p1 = Table(l,colWidths=widths,style=ts)
+                    ts.add('BACKGROUND', (0, 0), (0, -1), self._getSessionColor(sess_block))
+                    p1 = Table([["", p1]], colWidths=[0.2 * cm, None], style=ts)
+
                 res.append(p1)
                 if self._ttPDFFormat.showTitleSessionTOC():
-                    self._indexedFlowable[p1]={"text":escape(sessionSlot.getSession().getTitle()), "level":2}
-##                if self._useColors():
-##                    l = [["",p1]]
-##                    widths=[0.2*cm,None]
-##                    ts = deepcopy(colorts)
-##                    ts.add('BACKGROUND', (0, 0), (0,-1), self._getSessionColor(sessionSlot))
-##                    res.append(Table(l,colWidths=widths,style=ts))
-##                else:
-##                    res.append(p1)
-                #add session description
-                if sessionSlot.getSession().getDescription():
-                    text = "<i>%s</i>"%escape(sessionSlot.getSession().getDescription())
-                    p = Paragraph(text,self._styles["session_description"])
-                    res.append(p)
-                p2=Paragraph(conv,self._styles["conveners"])
+                    self._indexedFlowable[p1] = {'text': escape(sess_block.session.title), 'level': 2}
+
+                # add session description
+                if sess_block.session.description:
+                    text = '<i>{}</i>'.format(escape(sess_block.session.description))
+                    res.append(Paragraph(text, self._styles["session_description"]))
+
+                p2 = Paragraph(conv, self._styles["conveners"])
                 res.append(p2)
-                l=[]
+                l = []
                 ts = deepcopy(originalts)
-                if sessionSlot.getSession().getScheduleType()=="poster":
+                contribs = sorted([c for c in sess_block.contributions if not c.is_deleted],
+                                  key=attrgetter('timetable_entry.start_dt'))
+                if sess_block.session.is_poster:
                     if self._sortingCrit is not None:
-                        cl=[]
-                        for sEntry in sessionSlot.getSchedule().getEntries():
-                            if isinstance(sEntry,schedule.LinkedTimeSchEntry) and isinstance(sEntry.getOwner(),conference.Contribution):
-                                cl.append(sEntry.getOwner())
-                        f=filters.SimpleFilter(None,self._sortingCrit)
+                        cl = []
+                        for contrib in contribs:
+                            cl.append(contrib)
+                        f = filters.SimpleFilter(None, self._sortingCrit)
```

Comment:
I'm pretty sure those filters don't work on the new objects (and we probably don't need them anymore, they are pretty convoluted)